File: 2026_01/leetcode2026-01-31.py
from typing import List

class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        n = len(letters)
        l, r = 0, n-1
        while l<=r:
            mid = (l+r)//2
            if letters[mid] > target:
                r = mid-1
            else:
                l = mid+1
        return letters[l%n]

if __name__ == "__main__":
    s = Solution()
    letters = ["c","f","j"]
    target = "a"
    print(s.nextGreatestLetter(letters, target))

File: leetcode2026/2026_01/leetcode2026-01-31.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int n = letters.size();
        int l = 0, r = n-1;
        while (l<=r){
            int mid = (l+r)/2;
            if (letters[mid] > target){
                r = mid-1;
            }
            else{
                l = mid+1;
            }
        }
        return letters[l%n];
    }
};

int main(){
    Solution s;
    vector<char> letters = {'c', 'f', 'j'};
    char target = 'a';
    cout << s.nextGreatestLetter(letters, target) << endl;
}
